### PR TITLE
Match a few Sprite_3 procedures.

### PR DESCRIPTION
**Pull request by JaceCear**
* CopyOamBufferToOam
* sub_80058B4
* Add a few gFlags placeholders

### CODE REVIEW

File: src/sprite_3_2.c
Comment (freshollie) [inline src/sprite_3_2.c:1]: `malloc_oam.c`?

Comment (JaceCear) [reply src/sprite_3_2.c:1]: Maybe not a bad idea... but we'd have to have free in there as well for it to be the full module

Comment (freshollie) [reply src/sprite_3_2.c:1]: Where is the free function?

Comment (JaceCear) [reply src/sprite_3_2.c:1]: That is what I don't know! ^^

Comment (freshollie) [reply src/sprite_3_2.c:1]: So it's not near? In that case it's probably not it's own module. If there isn't a free function then I'm still ok with this being it's own module without the free function

**Review by freshollie — APPROVED** [review]

**Pull request merged**